presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctionsBase.java

```diff
@@ -0,0 +1,1121 @@
+
+        assertFunction("date_parse('1998 53', '%x %v')",
+                TimestampType.TIMESTAMP,
+                toTimestamp(new DateTime(1998, 12, 28, 0, 0, 0, 0, DATE_TIME_ZONE)));
+
+        assertFunction("date_parse('1.1', '%s.%f')",
+                TimestampType.TIMESTAMP,
+                toTimestamp(new DateTime(1970, 1, 1, 0, 0, 1, 100, DATE_TIME_ZONE)));
+        assertFunction("date_parse('1.01', '%s.%f')",
+                TimestampType.TIMESTAMP,
+                toTimestamp(new DateTime(1970, 1, 1, 0, 0, 1, 10, DATE_TIME_ZONE)));
+        assertFunction("date_parse('1.2006', '%s.%f')",
+                TimestampType.TIMESTAMP,
+                toTimestamp(new DateTime(1970, 1, 1, 0, 0, 1, 200, DATE_TIME_ZONE)));
+        assertFunction("date_parse('59.123456789', '%s.%f')",
+                TimestampType.TIMESTAMP,
+                toTimestamp(new DateTime(1970, 1, 1, 0, 0, 59, 123, DATE_TIME_ZONE)));
+
+        assertFunction("date_parse('0', '%k')",
+                TimestampType.TIMESTAMP,
+                toTimestamp(new DateTime(1970, 1, 1, 0, 0, 0, 0, DATE_TIME_ZONE)));
+
+        assertFunction("date_parse('28-JAN-16 11.45.46.421000 PM','%d-%b-%y %l.%i.%s.%f %p')",
+                TimestampType.TIMESTAMP,
+                toTimestamp(new DateTime(2016, 1, 28, 23, 45, 46, 421, DATE_TIME_ZONE)));
+        assertFunction("date_parse('11-DEC-70 11.12.13.456000 AM','%d-%b-%y %l.%i.%s.%f %p')",
+                TimestampType.TIMESTAMP,
+                toTimestamp(new DateTime(1970, 12, 11, 11, 12, 13, 456, DATE_TIME_ZONE)));
+        assertFunction("date_parse('31-MAY-69 04.59.59.999000 AM','%d-%b-%y %l.%i.%s.%f %p')",
+                TimestampType.TIMESTAMP,
+                toTimestamp(new DateTime(2069, 5, 31, 4, 59, 59, 999, DATE_TIME_ZONE)));
+
+        assertInvalidFunction("date_parse('', '%D')", "%D not supported in date format string");
+        assertInvalidFunction("date_parse('', '%U')", "%U not supported in date format string");
+        assertInvalidFunction("date_parse('', '%u')", "%u not supported in date format string");
+        assertInvalidFunction("date_parse('', '%V')", "%V not supported in date format string");
+        assertInvalidFunction("date_parse('', '%w')", "%w not supported in date format string");
+        assertInvalidFunction("date_parse('', '%X')", "%X not supported in date format string");
+
+        assertInvalidFunction("date_parse('3.0123456789', '%s.%f')", "Invalid format: \"3.0123456789\" is malformed at \"9\"");
+        assertInvalidFunction("date_parse('%Y-%M-%d', '')", "Both printing and parsing not supported");
+    }
+
+    @Test
+    public void testLocale()
+    {
+        Locale locale = Locale.JAPANESE;
+        Session localeSession = testSessionBuilder()
+                .setTimeZoneKey(TIME_ZONE_KEY)
+                .setLocale(locale)
+                .build();
+
+        FunctionAssertions localeAssertions = new FunctionAssertions(localeSession);
```

Comment:
`FunctionsAssertions` are closeable (and contain `LocalQueryRunner`) and should be closed.